realm/rfc/0001-backlink-query-syntax.md

```diff
@@ -0,0 +1,204 @@
+- Feature Name: Backlink query syntax
+- Start Date: 2016-05-30
+- RFC PR: 
+
+# Summary
+
+It should be possible to define and query backlinks in both `Realm` and 
+`DynamicRealm`. As a `DynamicRealm` can work without _any_ Realm model classes,
+it should be possible to query backlinks without them being defined there.
+
+# Motivation
+
+Backlinks are special in Realm. They are not defined explicitly in the schema,
+but are automatically created when defining Links and LinkLists (forward links).
+
+This creates some conflicts with the current query system as it rely on all 
+fields being defined by the schema. This is called *named backlinks*.
+
+This proposal introduces a new syntax for queries so it is possible to query
+across a backlink even though it hasn't been defined by a model class. This is
+called *unnamed backlinks*.
+
+Any design should ideally solve all these constraints:
+
+- Should support a list of backlinks.
+
+- Should support a single backlink, enforcing the constraint that only 1 object 
+  can point to it. This is not supported by Core, but it is a natural extension.
```

Comment:
Throwing an exception seems acceptable to me, especially considering it is a modelling constraint.
The primary driver is just ease of use though, e.g. not having everybody go through a `owners.size() > 0 ? owners.first() : null`. All the proposed solutions in the RFC support adding this in the future though. It does not have to be part of V1